diff --git a/media/gpu/vaapi/vaapi_wrapper.cc b/media/gpu/vaapi/vaapi_wrapper.cc
--- a/media/gpu/vaapi/vaapi_wrapper.cc
+++ b/media/gpu/vaapi/vaapi_wrapper.cc
@@ -3049,6 +3049,15 @@ bool VaapiWrapper::UploadVideoFrameToSurface(const VideoFrame& frame,
           UNSAFE_TODO(image_ptr + image.offsets[0]), image.pitches[0],
           UNSAFE_TODO(image_ptr + image.offsets[1]), image.pitches[1],
           visible_size.width(), visible_size.height());
+    } else if (frame.format() == PIXEL_FORMAT_NV12) {
+      ret = libyuv::NV12Copy(
+          frame.data(VideoFrame::Plane::kY),
+          frame.stride(VideoFrame::Plane::kY),
+          frame.data(VideoFrame::Plane::kUV),
+          frame.stride(VideoFrame::Plane::kUV),
+          UNSAFE_TODO(image_ptr + image.offsets[0]), image.pitches[0],
+          UNSAFE_TODO(image_ptr + image.offsets[1]), image.pitches[1],
+          visible_size.width(), visible_size.height());
     } else {
       LOG(ERROR) << "Unsupported pixel format: "
                  << VideoPixelFormatToString(frame.format());
